components/sendgrid/actions/delete-contacts/delete-contacts.js

```diff
@@ -0,0 +1,56 @@
+const sendgrid = require("../../sendgrid.app");
+const validate = require("validate.js");
+const common = require("../common");
+
+module.exports = {
+  key: "sendgrid-delete-contacts",
+  name: "Delete Contacts",
+  description: "Allows you to delete one or more contacts.",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    sendgrid,
+    deleteAllContacts: {
+      type: "boolean",
+      label: "Delete All Contacts?",
+      description: "This parameter allows you to delete all of your contacts. This can not be used with the `Ids` parameter.",
+      optional: true,
+    },
+    ids: {
+      type: "object",
+      label: "Ids",
+      description: "An array of contact IDs to delete.",
+      optional: true,
+    },
+  },
+  methods: {
+    ...common,
+  },
```

Comment:
When extending another component, make sure you refer to the correct nested attribute
```suggestion
  methods: {
    ...common.methods,
  },
```

However, in this case it shouldn't be necessary if you already do this at the beginning of the module definition:
```js
module.exports = {
  // This essentially sets every attribute defined in `common`
  // to be also defined in the current module
  ...common,
  props: {
    // Since re-defining the `props` attribute overrides the one previously
    // defined by `common`, we must explicitly say that we want those
    // as well.
    ...common.props,
    newProp: {},
  },
};
```